skynet_test.go

```diff
@@ -0,0 +1,53 @@
+package skynet
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestWalkDirectory(t *testing.T) {
+	testDir, err := ioutil.TempDir("", "testWalkDir")
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.RemoveAll(testDir)
+
+	subDir, err := ioutil.TempDir(testDir, "subDir")
+	if err != nil {
+		t.Error(err)
+	}
+
+	// lexical order to match filepath.Walk
+	testFiles := []string{
+		filepath.Join(testDir, "one"),
+		subDir,
+		filepath.Join(subDir, "foo"),
+		filepath.Join(testDir, "three"),
+		filepath.Join(testDir, "two"),
+	}
```

Comment:
Another option for testing would be to create a directory tree in a `testdata` folder, then walk that tree and verify it finds all of the files. This gives another way to verify the file.

Reply:
You mean have a `testdata` folder checked into the git repo? Yeah, that would work too. Let me know if you'd prefer I do it that way in this PR.